elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/SourceSettingsBase.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.elasticsearch
+
+import akka.util.JavaDurationConverters._
+import java.util.concurrent.TimeUnit
+
+import akka.stream.alpakka.elasticsearch.ElasticsearchConnectionSettings
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Configure Elastiscsearch/Opensearch sources.
+ *
+ */
+abstract class SourceSettingsBase[Version <: ApiVersionBase, S <: SourceSettingsBase[Version, S]] private[alpakka] (
+    val connection: ElasticsearchConnectionSettings,
+    val bufferSize: Int,
+    val includeDocumentVersion: Boolean,
+    val scrollDuration: FiniteDuration,
+    val apiVersion: Version
+) { this: S =>
+  def withConnection(value: ElasticsearchConnectionSettings): S = copy(connection = value)
+
+  def withBufferSize(value: Int): S = copy(bufferSize = value)
+
+  def withScrollDuration(value: FiniteDuration): S = copy(scrollDuration = value)
+
+  def withScrollDuration(value: java.time.Duration): S = copy(scrollDuration = value.asScala)
+
+  /**
+   * If includeDocumentVersion is true, '_version' is returned with the search-results
```

Comment:
These two links should probably be updated!  One is broken and the other is EOL.

Reply:
Updated the broken one :-)